<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="XSun" />

<meta name="date" content="2025-07-31" />

<title>Validation for initial submission</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rarevariant</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/myname/rarevariant">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Validation for initial submission</h1>
<h4 class="author">XSun</h4>
<h4 class="date">2025-07-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-08-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>rarevariant/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250619code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250619)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250619code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250619)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomxsun1229rarevarianttree7a0040f6e990d71f3b8143ca4d7b217f67700dd0targetblank7a0040fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/xsun1229/rarevariant/tree/7a0040f6e990d71f3b8143ca4d7b217f67700dd0" target="_blank">7a0040f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomxsun1229rarevarianttree7a0040f6e990d71f3b8143ca4d7b217f67700dd0targetblank7a0040fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/xsun1229/rarevariant/tree/7a0040f6e990d71f3b8143ca4d7b217f67700dd0" target="_blank">7a0040f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/mirage_lofdeconly.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/mirage_mis.Rmd
    Modified:   analysis/mirage_origin.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/mirage_origin.Rmd</code>) and
HTML (<code>docs/mirage_origin.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/rarevariant/blob/7a0040f6e990d71f3b8143ca4d7b217f67700dd0/analysis/mirage_origin.Rmd" target="_blank">7a0040f</a>
</td>
<td>
XSun
</td>
<td>
2025-07-31
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/rarevariant/7a0040f6e990d71f3b8143ca4d7b217f67700dd0/docs/mirage_origin.html" target="_blank">7a0040f</a>
</td>
<td>
XSun
</td>
<td>
2025-07-31
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/xsun1229/rarevariant/blob/ad9a90ad0e199efc9978328a15a741c484e33f1e/analysis/mirage_origin.Rmd" target="_blank">ad9a90a</a>
</td>
<td>
XSun
</td>
<td>
2025-07-31
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/xsun1229/rarevariant/ad9a90ad0e199efc9978328a15a741c484e33f1e/docs/mirage_origin.html" target="_blank">ad9a90a</a>
</td>
<td>
XSun
</td>
<td>
2025-07-31
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="extended-gene-sets" class="section level2">
<h2>Extended gene sets</h2>
<pre class="r"><code>genesets &lt;- readRDS(&quot;/project/xinhe/xsun/rare_variants/5.revision/data/genesets/genesets_summary.RDS&quot;)

mirage &lt;- readRDS(&quot;/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/gene_pp_bf_posadded_origin.RDS&quot;)

num_topgene &lt;- sum(mirage$PP &gt; 0.5)
mirage_genes &lt;- mirage$Gene[mirage$PP &gt; 0.5]
all_genes &lt;- mirage$Gene

summary &lt;- c()
for (i in 1:length(genesets)){
  
  genes_inset &lt;- genesets[[i]]
  
  num_mirage_gene_in_geneset &lt;- sum(mirage_genes %in% genes_inset)
  
  # Create categories
  mirage_in_set     &lt;- sum(mirage_genes %in% genes_inset)
  mirage_not_in_set &lt;- length(mirage_genes) -  mirage_in_set 
  non_mirage_in_set &lt;- sum(setdiff(all_genes, mirage_genes) %in% genes_inset)
  non_mirage_not_in_set &lt;- length(setdiff(all_genes, mirage_genes)) -  non_mirage_in_set
  
  gene_detail &lt;- mirage_genes[mirage_genes %in% genes_inset]
  
  # Build 2x2 matrix
  contingency_table &lt;- matrix(c(
    mirage_in_set,
    mirage_not_in_set,
    non_mirage_in_set,
    non_mirage_not_in_set
  ), nrow = 2, byrow = TRUE)
  
  colnames(contingency_table) &lt;- c(&quot;In_Set&quot;, &quot;Not_In_Set&quot;)
  rownames(contingency_table) &lt;- c(&quot;MIRAGE&quot;, &quot;Non_MIRAGE&quot;)
  
  # Perform Fisher&#39;s Exact Test
  fisher_res &lt;- fisher.test(contingency_table)

  tmp &lt;- c(names(genesets[i]),length(genes_inset), mirage_in_set, paste0(round(mirage_in_set/length(mirage_genes) *100,2), &quot;%&quot;), 
           round(fisher_res$p.value,4), round(fisher_res$estimate,4),paste0(gene_detail,collapse = &quot;,&quot;))
  summary &lt;- rbind(summary, tmp)
  
}

summary &lt;- as.data.frame(summary)
colnames(summary) &lt;- c(&quot;gene sets&quot;, &quot;#of genes in gene set&quot;, &quot;#of mirage gene in gene set&quot;, &quot;pct of mirage gene in gene set&quot;, &quot;FET p&quot;, &quot;FET OR&quot;, &quot;MIRAGE genes in gene set&quot;)

rownames(summary) &lt;- NULL

rownames(summary)[rownames(summary) == &quot;High&quot;] &lt;- &quot;TADA&quot;

DT::datatable(summary,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;FET results for extended gene set&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-de8cc9eb1fa1fdf58fa2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-de8cc9eb1fa1fdf58fa2">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">FET results for extended gene set<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20"],["ID gene","High","PSD","FMRP","AutismKB","constraint gene","RVIS","Haploinsuff","SCZ gene","SFARI_gene","SCZ_SCHEMA","DD","ADHD","Anxiety","Bipolar","OCD","MDD","PTSD","OCD_smaller","PTSD_smaller"],["252","120","661","783","171","1003","846","1440","1796","769","32","285","76","158","36","251","296","415","25","47"],["0","2","5","7","0","5","6","7","8","7","0","1","1","0","0","0","2","0","0","0"],["0%","11.11%","27.78%","38.89%","0%","27.78%","33.33%","38.89%","44.44%","38.89%","0%","5.56%","5.56%","0%","0%","0%","11.11%","0%","0%","0%"],["1","0.007","4e-04","0","1","0.0016","1e-04","3e-04","2e-04","0","1","0.2347","0.0745","1","1","1","0.0216","1","1","1"],["0","17.8295","10.2269","14.0487","0","7.2479","10.1216","7.2545","7.2626","14.699","0","3.9427","13.8286","0","0","0","9.7591","0","0","0"],["","ATP2B2,SYNGAP1","PLXNA1,MYH10,AHNAK,ATP2B2,SYNGAP1","PLXNA1,MYH10,PTPRF,NAV3,ARAP2,ATP2B2,SYNGAP1","","PLXNA1,MYH10,SF3B3,ATP2B2,SYNGAP1","PLXNA1,MYH10,PTPRF,NAV3,AHNAK,ATP2B2","PLXNA1,MYH10,PTPRF,NAV3,ARAP2,AHNAK,MYO5C","PLXNA1,MYH10,PTPRF,NAV3,ARAP2,AHNAK,ATP2B2,SYNGAP1","MYH10,NAV3,AHNAK,IL1R2,ATP2B2,MYO5C,KATNAL1","","SYNGAP1","PTPRF","","","","PCYOX1L,NAV3","","",""]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>gene sets<\/th>\n      <th>#of genes in gene set<\/th>\n      <th>#of mirage gene in gene set<\/th>\n      <th>pct of mirage gene in gene set<\/th>\n      <th>FET p<\/th>\n      <th>FET OR<\/th>\n      <th>MIRAGE genes in gene set<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="enrichment-in-other-gene-sets" class="section level2">
<h2>Enrichment in other gene sets</h2>
<p><a href="https://pubmed.ncbi.nlm.nih.gov/31981491/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/31981491/</a> This paper
published several data sets we can use. The gene sets are gtex tissue
specific genes, prenatal vs. postnatal gene sets.</p>
<p>Some gene sets were downloaded here <a
href="https://github.com/BreenMS/Rapid-gene-set-enrichment"
class="uri">https://github.com/BreenMS/Rapid-gene-set-enrichment</a></p>
<pre class="r"><code>gene_pp_bf_posadded_origin &lt;- readRDS(&quot;/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/gene_pp_bf_posadded_origin.RDS&quot;)

get_fisher_enrichment &lt;- function(category_genes, category_name) {
  # Restrict all sets to background to be consistent
  bg &lt;- background_genes
  highpip &lt;- intersect(highpip_genes, bg)
  category &lt;- intersect(category_genes, bg)
  
  # Counts for contingency table
  a &lt;- sum(highpip %in% category)              # HighPIP &amp; In Category
  b &lt;- sum(highpip %notin% category)           # HighPIP &amp; Not in Category
  c &lt;- sum((bg %notin% highpip) &amp; (bg %in% category))  # Not HighPIP &amp; In Category
  d &lt;- sum((bg %notin% highpip) &amp; (bg %notin% category)) # Not HighPIP &amp; Not in Category
  
  contingency_table &lt;- matrix(c(a, b, c, d), nrow = 2,
                              dimnames = list(
                                HighPIP = c(&quot;Yes&quot;, &quot;No&quot;),
                                InCategory = c(&quot;Yes&quot;, &quot;No&quot;)
                              ))
  
  fisher_result &lt;- fisher.test(contingency_table, alternative = &quot;greater&quot;)
  
  cat(&quot;\n&quot;, category_name, &quot;\n&quot;)
  print(contingency_table)
  print(fisher_result)
}</code></pre>
<div id="gtex-brain-tissue-specific-genes" class="section level3">
<h3>GTEx brain tissue specific genes</h3>
<p>We did Fisher exact test to check if the genes with PP&gt;0.5 are
enriched in the tissue-specific gene sets.</p>
<pre class="r"><code>merged_df &lt;- readRDS(&quot;/project/xinhe/xsun/rare_variants/5.revision/results/validation_gtex_brain_origin.RDS&quot;)

DT::datatable(merged_df,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39; Brain tissue upregulated genes and overlap with MIRAGE genes&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-48defda8782ef968da74" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-48defda8782ef968da74">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\"> Brain tissue upregulated genes and overlap with MIRAGE genes<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12"],["Brain_Amygdala.up_1996","Brain_Anterior_cingulate_cortex_BA24.up_2523","Brain_Caudate_basal_ganglia.up_2327","Brain_Cerebellar_Hemisphere.up_6857","Brain_Cerebellum.up_6616","Brain_Cortex.up_2969","Brain_Frontal_Cortex_BA9.up_3421","Brain_Hippocampus.up_2170","Brain_Hypothalamus.up_2782","Brain_Nucleus_accumbens_basal_ganglia.up_2754","Brain_Putamen_basal_ganglia.up_1867","Brain_Substantia_nigra.up_1907"],["ATP2B2","NAV3,ATP2B2","ARAP2,ATP2B2","MYH10,PCYOX1L,LMNB1,NAV3,ATP2B2,SYNGAP1,MYO5C,KATNAL1","MYH10,PCYOX1L,LMNB1,NAV3,ATP2B2,PAN2,SYNGAP1,MYO5C","PLXNA1,NAV3,ATP2B2,SYNGAP1","MYH10,PCYOX1L,NAV3,ARAP2,ATP2B2,SYNGAP1","NAV3,ATP2B2,SYNGAP1","ATP2B2","ARAP2,ATP2B2,SYNGAP1","ARAP2,ATP2B2","ATP2B2"],[0.800427126864261,0.589735777873888,0.546696130416659,0.117970594288528,0.0986502182311021,0.184473367386715,0.037156634238465,0.232366137222555,0.895790734320166,0.354285088510465,0.428695561738463,0.784915319109602]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>tissue<\/th>\n      <th>mirage_gene_overlap<\/th>\n      <th>enrich_p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>merged_df &lt;- readRDS(&quot;/project/xinhe/xsun/rare_variants/5.revision/results/validation_gtex_nonbrain_origin.RDS&quot;)


DT::datatable(merged_df,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39; non-Brain tissue upregulated genes and overlap with MIRAGE genes&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-8162b872ed07ea29f25d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8162b872ed07ea29f25d">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\"> non-Brain tissue upregulated genes and overlap with MIRAGE genes<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39"],["Adipose_Subcutaneous.up_2148","Adipose_Visceral_Omentum.up_1828","Adrenal_Gland.up_1656","Artery_Aorta.up_2673","Artery_Coronary.up_2000","Artery_Tibial.up_2567","Bladder.up_608","Breast_Mammary_Tissue.up_2059","Cells_Transformed_fibroblasts.up_4268","Cervix_Ectocervix.up_60","Cervix_Endocervix.up_363","Colon_Sigmoid.up_1649","Colon_Transverse.up_1715","Esophagus_Gastroesophageal_Junction.up_1329","Esophagus_Mucosa.up_2327","Esophagus_Muscularis.up_1312","Fallopian_Tube.up_227","Heart_Atrial_Appendage.up_766","Heart_Left_Ventricle.up_568","Kidney_Cortex.up_1347","Liver.up_1568","Lung.up_4357","Minor_Salivary_Gland.up_1922","Muscle_Skeletal.up_1141","Nerve_Tibial.up_5558","Ovary.up_5788","Pancreas.up_808","Pituitary.up_5270","Prostate.up_3415","Skin_Not_Sun_Exposed_Suprapubic.up_2690","Skin_Sun_Exposed_Lower_leg.up_2799","Small_Intestine_Terminal_Ileum.up_2823","Spleen.up_4843","Stomach.up_1164","Testis.up_12646","Thyroid.up_5868","Uterus.up_5588","Vagina.up_2791","Whole_Blood.up_2262"],["MYO5C","IL1R2","","PLXNA1,MYH10,BTG3","PLXNA1,MYH10","MYH10,KATNAL1","PTPRF","PTPRF,MYO5C","PLXNA1,MYH10,LMNB1,NAV3,SF3B3,KATNAL1","SF3B3,BTG3","SF3B3,SYNGAP1","KATNAL1","PTPRF,LMNB1,IL1R2","KATNAL1","PTPRF,LMNB1,ARAP2,SF3B3,IL1R2","KATNAL1","MYH10,BTG3","","","","ATP2B2","PLXNA1,MYH10,LMNB1,ARAP2,IL1R2,BTG3,MYO5C","PTPRF,MYO5C","","PCYOX1L,PTPRF,NAV3,SF3B3,PAN2,KATNAL1","PLXNA1,MYH10,PCYOX1L,NAV3,SF3B3,SYNGAP1","MYO5C","MYH10,PCYOX1L,SYNGAP1,MYO5C","PTPRF,MYO5C","PTPRF,ARAP2,IL1R2,BTG3","PTPRF,ARAP2,SF3B3,IL1R2,BTG3","LMNB1","LMNB1,NAV3,IL1R2","","PCYOX1L,LMNB1,BTG3,PAN2,KATNAL1","ARAP2,BTG3,PAN2,MYO5C","MYH10,PTPRF,SF3B3,BTG3,SYNGAP1,KATNAL1","PTPRF,SF3B3,IL1R2,BTG3","LMNB1,IL1R2"],[0.822570234698893,0.767073802006118,1,0.339064580163417,0.465725184198872,0.60657970719121,0.38347606322542,0.476893940128349,0.0986274205758246,0.000988979788141114,0.0322636057809055,0.73077890758803,0.138867123318176,0.647756078955911,0.0285977071356788,0.644202654643165,0.0132450893562763,1,1,1,0.714086952377232,0.0361582128332801,0.440795376102089,1,0.237004020890454,0.266474209907661,0.468955147478601,0.590354118017189,0.754962560659449,0.142806180095926,0.0556036527939014,0.898272249832074,0.738477938742159,1,0.992556350244961,0.685858895405902,0.242394928347952,0.158912530180113,0.524158529459524]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>tissue<\/th>\n      <th>mirage_gene_overlap<\/th>\n      <th>enrich_p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="prenatal-postnatal-unbiased-gene-sets" class="section level3">
<h3>prenatal, postnatal, unbiased gene sets</h3>
<p>We did Fisher exact test to check if the genes with PP&gt;0.5 are
enriched in the prenatal/postnatal/unbiased gene sets.</p>
<pre class="r"><code>braineffect &lt;-  readxl::read_excel(&quot;/project/xinhe/xsun/rare_variants/5.revision/data/brain_de/BrainSpan_FetalEffect.xlsx&quot;)

prenatal &lt;- braineffect[braineffect$FetalEffect == &quot;Prenatal&quot;,]
unbiased &lt;- braineffect[braineffect$FetalEffect == &quot;Unbiased&quot;,]
postnatal &lt;- braineffect[braineffect$FetalEffect == &quot;Postnatal&quot;,]

print(paste0(&quot;Prenatal gene number =&quot;, nrow(prenatal)))</code></pre>
<pre><code>[1] &quot;Prenatal gene number =8696&quot;</code></pre>
<pre class="r"><code>print(paste0(&quot;Postnatal gene number =&quot;, nrow(postnatal)))</code></pre>
<pre><code>[1] &quot;Postnatal gene number =7203&quot;</code></pre>
<pre class="r"><code>print(paste0(&quot;Unbiased gene number =&quot;, nrow(unbiased)))</code></pre>
<pre><code>[1] &quot;Unbiased gene number =10596&quot;</code></pre>
<pre class="r"><code>print(&quot;Genes with PP &gt; 0.5&quot;)</code></pre>
<pre><code>[1] &quot;Genes with PP &gt; 0.5&quot;</code></pre>
<pre class="r"><code>highpip &lt;- gene_pp_bf_posadded_origin[gene_pp_bf_posadded_origin$PP &gt; 0.5,]

# Define sets
highpip_genes &lt;- highpip$Gene
prenatal_genes &lt;- prenatal$GeneSymbol
postnatal_genes &lt;- postnatal$GeneSymbol
unbiased_genes &lt;- unbiased$GeneSymbol

background_genes &lt;- gene_pp_bf_posadded_origin$Gene


# Define `%notin%` for convenience
`%notin%` &lt;- Negate(`%in%`)

print(&quot;Prenatal&quot;)</code></pre>
<pre><code>[1] &quot;Prenatal&quot;</code></pre>
<pre class="r"><code>print(highpip$Gene[highpip$Gene %in% prenatal$GeneSymbol])</code></pre>
<pre><code>[1] &quot;PLXNA1&quot; &quot;PTPRF&quot;  &quot;LMNB1&quot;  &quot;NAV3&quot;   &quot;SF3B3&quot;  &quot;MFSD10&quot;</code></pre>
<pre class="r"><code>get_fisher_enrichment(prenatal_genes, &quot;Prenatal&quot;)</code></pre>
<pre><code>
 Prenatal 
       InCategory
HighPIP Yes    No
    Yes   6  4935
    No   12 11598

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.4601
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 0.4352797       Inf
sample estimates:
odds ratio 
  1.175043 </code></pre>
<pre class="r"><code>print(&quot;Postnatal&quot;)</code></pre>
<pre><code>[1] &quot;Postnatal&quot;</code></pre>
<pre class="r"><code>print(highpip$Gene[highpip$Gene %in% postnatal$GeneSymbol])</code></pre>
<pre><code>[1] &quot;PCYOX1L&quot; &quot;ARAP2&quot;   &quot;AHNAK&quot;   &quot;ATP2B2&quot;  &quot;MYO5C&quot;  </code></pre>
<pre class="r"><code>get_fisher_enrichment(postnatal_genes, &quot;Postnatal&quot;)</code></pre>
<pre><code>
 Postnatal 
       InCategory
HighPIP Yes    No
    Yes   5  4451
    No   13 12082

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.5565
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 0.3574851       Inf
sample estimates:
odds ratio 
  1.044018 </code></pre>
<pre class="r"><code>print(&quot;Unbiased&quot;)</code></pre>
<pre><code>[1] &quot;Unbiased&quot;</code></pre>
<pre class="r"><code>print(highpip$Gene[highpip$Gene %in% unbiased$GeneSymbol])</code></pre>
<pre><code>[1] &quot;MYH10&quot;   &quot;BTG3&quot;    &quot;PAN2&quot;    &quot;SYNGAP1&quot; &quot;KATNAL1&quot;</code></pre>
<pre class="r"><code>get_fisher_enrichment(unbiased_genes, &quot;Unbiased&quot;)</code></pre>
<pre><code>
 Unbiased 
       InCategory
HighPIP Yes    No
    Yes   5  3771
    No   13 12762

    Fisher&#39;s Exact Test for Count Data

data:  contingency_table
p-value = 0.3938
alternative hypothesis: true odds ratio is greater than 1
95 percent confidence interval:
 0.4456408       Inf
sample estimates:
odds ratio 
  1.301611 </code></pre>
</div>
<div id="single-cell-expression-data" class="section level3">
<h3>Single cell expression data,</h3>
<p>To assess whether the ASD genes are enriched in specific cell types
(clusters), we used a background set of 7,867 genes expressed in the
experiment. For each cell type, they built a 2×2 table comparing the
number of ASD genes expressed versus not expressed, and the number of
non-ASD genes expressed versus not expressed. Enrichment was evaluated
using odds ratios and Fisher’s exact test.</p>
<p>Within each cell type cluster, a gene was considered expressed if one
or more of its transcripts were detected in 25% or more cells</p>
<pre class="r"><code>fisher_res &lt;- readRDS(&quot;/project/xinhe/xsun/rare_variants/5.revision/results/validation_scrna_fet_origin.RDS&quot;)

DT::datatable(fisher_res,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Fisher results for the ASD genes&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-ce92b5a4e78853dbdd81" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ce92b5a4e78853dbdd81">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Fisher results for the ASD genes<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19"],["C1-Striatal Interneurons","C2-Early + Late Excitation Neurons (V1 enriched)","C3-Early Excitatory Neurons","C4-Astrocytes and OPCs","C6-MGE derived cortical interneurons","C7-Dividing Intermediate Progenitor Cells (IPCs)","C8-MGE IPCs","C9-Endothelial","C10-Radial Glia","C11-MGE RG","C13-Newborn Neurons","C15-CGE derived interneurons","C16-Early + Late Excitation Neurons (PFC enriched)","C17-IPCs","C18-Newborn Neurons","C19-Microglia","C20-Choroid Plexus","C23-Newborn MGE Interneurons","C25-Dividing RG"],["0.162744346524004","0.364663348622402","0.416995785215401","0.324743952646109","0.286436469941069","0.255490461435439","0.536625268938738","0.164616727166947","0.35275833911138","0.416010000446189","0.388892188344791","0.320975124055687","0.322276119276254","0.298388673289049","0.193578992184883","0.110561891394882","0.188607003468394","0.42344290731236","0.232822472144174"],["0.0571439718956669","0.0531082766930477","0.0887140046706101","0.0315667314875622","0.119687299336633","0.00510885053132814","0.242837204614029","0.000270920350945831","0.0450705752624585","0.149308037965016","0.0520946025540566","0.185033727103428","0.0267718601923046","0.0526636332214306","0.0144245461215282","0.0105246599917142","0.00358477245354417","0.396416206311919","0.00286397421681922"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>cluster<\/th>\n      <th>odds ratio<\/th>\n      <th>fisher-p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cluster_member &lt;- readRDS(&quot;/project/xinhe/xsun/rare_variants/5.revision/results/validation_scrna_expr_binary_origin.RDS&quot;)

DT::datatable(cluster_member,caption = htmltools::tags$caption( style = &#39;caption-side: left; text-align: left; color:black;  font-size:150% ;&#39;,&#39;Cluster membership for the ASD genes&#39;),options = list(pageLength = 10) )</code></pre>
<div id="htmlwidget-bae1458cb9fae638aeff" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bae1458cb9fae638aeff">{"x":{"filter":"none","vertical":false,"caption":"<caption style=\"caption-side: left; text-align: left; color:black;  font-size:150% ;\">Cluster membership for the ASD genes<\/caption>","data":[["PLXNA1","MYH10","PCYOX1L","PTPRF","LMNB1","NAV3","ARAP2","AHNAK","SF3B3","IL1R2","ATP2B2","BTG3","MFSD10","OR52A5","PAN2","SYNGAP1","MYO5C","KATNAL1"],[0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[1,1,0,1,0,1,0,0,1,0,0,0,0,0,0,0,0,1],[1,1,0,1,0,1,0,0,1,0,1,0,0,0,0,0,0,1],[0,1,0,1,0,0,1,0,1,0,0,1,0,0,0,0,0,0],[0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0],[0,1,0,1,1,0,0,0,1,0,0,1,0,0,0,0,0,1],[0,1,0,1,1,0,0,0,1,0,0,1,0,0,0,0,0,1],[0,1,0,0,0,1,1,1,1,0,0,0,0,0,0,0,0,0],[0,1,0,1,0,1,1,0,1,0,0,1,0,0,0,0,0,1],[0,1,0,1,0,0,0,0,1,0,0,1,0,0,0,0,0,0],[1,1,0,1,0,0,0,0,1,0,1,1,0,0,0,0,0,1],[0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0],[1,1,0,1,0,1,0,0,1,0,0,0,0,0,0,0,0,1],[0,1,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0],[0,1,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0],[0,1,0,0,0,0,0,0,1,0,0,1,0,0,0,0,0,0],[0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0],[0,1,0,1,1,1,1,0,1,0,0,1,0,0,1,0,0,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>C1-Striatal Interneurons<\/th>\n      <th>C2-Early + Late Excitation Neurons (V1 enriched)<\/th>\n      <th>C3-Early Excitatory Neurons<\/th>\n      <th>C4-Astrocytes and OPCs<\/th>\n      <th>C6-MGE derived cortical interneurons<\/th>\n      <th>C7-Dividing Intermediate Progenitor Cells (IPCs)<\/th>\n      <th>C8-MGE IPCs<\/th>\n      <th>C9-Endothelial<\/th>\n      <th>C10-Radial Glia<\/th>\n      <th>C11-MGE RG<\/th>\n      <th>C13-Newborn Neurons<\/th>\n      <th>C15-CGE derived interneurons<\/th>\n      <th>C16-Early + Late Excitation Neurons (PFC enriched)<\/th>\n      <th>C17-IPCs<\/th>\n      <th>C18-Newborn Neurons<\/th>\n      <th>C19-Microglia<\/th>\n      <th>C20-Choroid Plexus<\/th>\n      <th>C23-Newborn MGE Interneurons<\/th>\n      <th>C25-Dividing RG<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":10,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
[1] C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12       cellranger_1.1.0  pillar_1.9.0      compiler_4.2.0   
 [5] bslib_0.3.1       later_1.3.0       jquerylib_0.1.4   git2r_0.30.1     
 [9] workflowr_1.7.0   tools_4.2.0       digest_0.6.29     jsonlite_1.8.0   
[13] evaluate_0.15     lifecycle_1.0.4   tibble_3.2.1      pkgconfig_2.0.3  
[17] rlang_1.1.2       cli_3.6.1         rstudioapi_0.13   crosstalk_1.2.0  
[21] yaml_2.3.5        xfun_0.41         fastmap_1.1.0     stringr_1.5.1    
[25] knitr_1.39        fs_1.5.2          vctrs_0.6.5       sass_0.4.1       
[29] htmlwidgets_1.5.4 rprojroot_2.0.3   DT_0.22           glue_1.6.2       
[33] R6_2.5.1          fansi_1.0.3       readxl_1.4.0      rmarkdown_2.25   
[37] magrittr_2.0.3    whisker_0.4       promises_1.2.0.1  htmltools_0.5.2  
[41] httpuv_1.6.5      utf8_1.2.2        stringi_1.7.6    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
