---
title: "MIRAGE results -- missense annotation updated"
author: "XSun"
date: "2025-06-23"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

```{r, warning=F, message=FALSE}
library(ggplot2)
library(dplyr)
library(ggrepel)
library(data.table)
library(tidyr)
library(gridExtra)
library(VennDiagram)
library(tidyverse)

mirage_res <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/mirage_misupdated.RDS")

#cat <- c("high LOEUF oe", "med LOEUF oe", "low LOEUF oe", "likely pathogenic", "ambiguous", "likely benign")
cat <- c("highPLI", "medPLI", "lowPLI", "likely pathogenic", "ambiguous", "likely benign")
cat_old <- c("highPLI", "medPLI", "lowPLI", "highMPC", "medMPC", "lowMPC")

# Function to test enrichment with Fisher's exact test
get_fisher_enrichment <- function(category_genes, category_name) {
  # Restrict all sets to background to be consistent
  bg <- background_genes
  highpip <- intersect(highpip_genes, bg)
  category <- intersect(category_genes, bg)
  
  # Counts for contingency table
  a <- sum(highpip %in% category)              # HighPIP & In Category
  b <- sum(highpip %notin% category)           # HighPIP & Not in Category
  c <- sum((bg %notin% highpip) & (bg %in% category))  # Not HighPIP & In Category
  d <- sum((bg %notin% highpip) & (bg %notin% category)) # Not HighPIP & Not in Category
  
  contingency_table <- matrix(c(a, b, c, d), nrow = 2,
                              dimnames = list(
                                HighPIP = c("Yes", "No"),
                                InCategory = c("Yes", "No")
                              ))
  
  fisher_result <- fisher.test(contingency_table, alternative = "greater")
  
  cat("\n", category_name, "\n")
  print(contingency_table)
  print(fisher_result)
}

```

We are addressing the reviews' comments:

```
R1:
3)    More rigorous prediction scores (i.e., REVEL, Alpha-missense) can be included for the stratification of missense variants.


R2:
2.  	The authors used the MPC score. I would argue this is now outdated and there are much better tools for assessing missense constaint and pathogenicity.
```


# Updating missense annotation using Alpha-missense score

Alpha-missense score download: https://console.cloud.google.com/storage/browser/dm_alphamissense;tab=objects?pli=1&inv=1&invt=Ab0dvQ&prefix=&forceOnObjectsSortingFiltering=false

Paper: https://www.science.org/doi/10.1126/science.adg7492

The authors compared with alphamissense with other missense annotations: https://www.ebi.ac.uk/training/online/courses/alphafold/classifying-the-effects-of-missense-variants-using-alphamissense/introducing-missense-variants-and-alphamissense/

```
zcat /project/xinhe/xsun/rare_variants/5.revision/data/alphamis/AlphaMissense_hg38.tsv.gz |head
# Copyright 2023 DeepMind Technologies Limited
#
# Licensed under CC BY-NC-SA 4.0 license
CHROM  POS     REF     ALT     genome  uniprot_id      transcript_id   protein_variant am_pathogenicity        am_class
chr1    69094   G       T       hg38    Q8NH21  ENST00000335137.4       V2L     0.2937  likely_benign
chr1    69094   G       C       hg38    Q8NH21  ENST00000335137.4       V2L     0.2937  likely_benign
chr1    69094   G       A       hg38    Q8NH21  ENST00000335137.4       V2M     0.3296  likely_benign
chr1    69095   T       C       hg38    Q8NH21  ENST00000335137.4       V2A     0.2609  likely_benign
chr1    69095   T       A       hg38    Q8NH21  ENST00000335137.4       V2E     0.2922  likely_benign
```

```
| Category          | Min pathogenicity | Max pathogenicity | Total Variants |
|-------------------|-----------|-----------|----------------|
| likely_pathogenic | 0.5640    | 1.0000    | 22,770,557     |
| likely_benign     | 0.0001    | 0.3400    | 40,917,351     |
| ambiguous         | 0.3400    | 0.5640    | 8,009,648      |
```

More details for annotaion, see https://xsun1229.github.io/rarevariant/variant_annot.html

# Overview of MIRAGE input

```{r fig.height=4, fig.width=4 , warning=F, message=FALSE}

input <- fread("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_20231013.txt")
print("Old mirage input:")

sprintf("Number of rare-variants = %s",nrow(input))
old_mirage <- table(input$group.index)[c("1","2","3","4","5","6")]
names(old_mirage) <- c("highPLI", "medPLI", "lowPLI", "highMPC", "medMPC", "lowMPC")
print(old_mirage)

print("Missense updated mirage input:")
mirage_input_mis_updated <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_misupdated.RDS")
new_mirage <- table(mirage_input_mis_updated$group.index)[c("1","2","3","4","5","6")]
names(new_mirage) <- cat
sprintf("Number of rare-variants = %s",nrow(mirage_input_mis_updated))
print(new_mirage)


highmpc <- input[input$group.index == 4,]
medmpc <- input[input$group.index == 5,]
lowmpc <- input[input$group.index == 6,]

lp <- mirage_input_mis_updated[mirage_input_mis_updated$group.index == 4,]
ab <- mirage_input_mis_updated[mirage_input_mis_updated$group.index == 5,]
lb <- mirage_input_mis_updated[mirage_input_mis_updated$group.index == 6,]

overlap_highlp <- sum(highmpc$Variant %in% lp$Variant)
overlap_medab <- sum(medmpc$Variant %in% ab$Variant)
overlap_lowlb <- sum(lowmpc$Variant %in% lb$Variant)


grid.newpage()
venn.plot <- draw.pairwise.venn(
  area1 = nrow(highmpc),         # size of A
  area2 = nrow(lp),         # size of B
  cross.area = overlap_highlp,   # size of overlap A ∩ B
  category = c("highmpc", "likely_pathogenic"),
  fill = c("skyblue", "pink1"),
  alpha = 0.5,
  cex = 2,
  cat.cex = 2,
  cat.pos = c(-20, 20)
)


```

```{r fig.height=4, fig.width=4 , warning=F, message=FALSE}

grid.newpage()
venn.plot <- draw.pairwise.venn(
  area1 = nrow(medmpc),         # size of A
  area2 = nrow(ab),         # size of B
  cross.area = overlap_medab,   # size of overlap A ∩ B
  category = c("medmpc", "ambiguous "),
  fill = c("skyblue", "pink1"),
  alpha = 0.5,
  cex = 2,
  cat.cex = 2,
  cat.pos = c(-20, 20)
)

```

```{r fig.height=4, fig.width=4 , warning=F, message=FALSE}
grid.newpage()

venn.plot <- draw.pairwise.venn(
  area1 = nrow(lowmpc),         # size of A
  area2 = nrow(lb),         # size of B
  cross.area = overlap_lowlb,   # size of overlap A ∩ B
  category = c("lowmpc", "likely_benign"),
  fill = c("skyblue", "pink1"),
  alpha = 0.5,
  cex = 2,
  cat.cex = 2,
  cat.pos = c(-20, 20)
)


# venn.plot <- draw.pairwise.venn(
#   area1 = nrow(highmpc),         # size of A
#   area2 = nrow(lb),         # size of B
#   cross.area = overlap_highlp,   # size of overlap A ∩ B
#   category = c("highmpc", "likely_benign"),
#   fill = c("skyblue", "pink1"),
#   alpha = 0.5,
#   cex = 2,
#   cat.cex = 2,
#   cat.pos = c(-20, 20)
# )
# 
# # To plot in RStudio Viewer
# grid.newpage()
# grid.draw(venn.plot)
```


# Parameter estimation

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}
est <- mirage_res$eta.est[c("1","2","3","4","5","6")]
names(mirage_res$eta.pvalue) <- names(mirage_res$eta.est)
pval <- mirage_res$eta.pvalue[c("1","2","3","4","5","6")]


order <- c(1:length(est))

df <- data.frame(est = est, pval = pval, cat = cat, order=order)
df <- df[sort(df$order,decreasing = T),]
df$cat <- factor(df$cat, levels = cat)

df$fill <- ifelse(df$order <= 3, "orange", "steelblue")
df$legend <- ifelse(df$order <= 3, "LoF", "Missense")

# Convert the legend column to a factor and set the levels
df$legend <- factor(df$legend, levels = c("LoF", "Missense"))

p <- ggplot(df, aes(x = cat, y = est, fill = legend)) +
  geom_bar(stat = "identity", width = 0.6) + # Use the fill aesthetic
  geom_text(aes(label = ifelse(pval < 0.05, "*", "")), vjust = 0, size = 6) +
  annotate("text", x = 6, y = 0.9, label = "*: p-value < 0.05", hjust = "right", size = 6) +
  theme_minimal() +
  theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
    axis.text.y = element_text(size = 12),
    legend.title = element_blank() # Hide the legend title
  ) +
  scale_y_continuous(breaks = seq(0, 1, 0.2)) +
  scale_fill_manual(values = c("LoF" = "orange", "Missense" = "steelblue")) # Define manual colors for the legend

print(p)

```


# Gene-level results

## Posterior probability, FDR 

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

gene_pip <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/gene_pp_bf_posadded_misupdated.RDS")

DT::datatable(gene_pip,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Gene-level results'),options = list(pageLength = 10) )

```


## Manhattan plot

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

pp_threshold <- 0.7

don <- gene_pip %>% 
  
  # Compute chromosome size
  group_by(chr) %>% 
  summarise(chr_len=max(pos)) %>% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(gene_pip, ., by=c("chr"="chr")) %>%
  
  # Add a cumulative position of each SNP
  arrange(chr, pos) %>%
  mutate( poscum=pos+tot) %>%
  mutate( is_annotate=ifelse(Gene %in% gene_pip$Gene[gene_pip$post_prob > pp_threshold], "yes", "no"))

# Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
axisdf = don %>% group_by(chr) %>% summarize(center=( max(poscum) + min(poscum) ) / 2 )

manhplot <- ggplot(don, aes(x=poscum, y=post_prob)) +
  
  # Show all points
  geom_point( aes(color=as.factor(chr)), alpha=2, size=3) +
  scale_color_manual(values = rep(c("grey", "steelblue"), 22 )) +
  
  # custom X axis:
  scale_x_continuous( label = axisdf$chr, breaks= axisdf$center ) +
  #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  
  ylab("Posterior Probability") +
  #ylim(0,max(log10(don$BF)) + 2) +
  ylim(0,1) +
  xlab("Chromosome") +
  
  geom_hline(yintercept = pp_threshold, col = "red") +
  annotate("text", x = Inf, y = pp_threshold, label = sprintf("PP = %s", pp_threshold), hjust = 1.1, vjust = -1, color = "red", size = 7) +
  
  # Custom the theme:
  theme_bw() +
  #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
  theme( 
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_text(size = 28),
    axis.text.x = element_text(size = 26, color = "black"),
    axis.title.y = element_text(size = 28),
    axis.text.y = element_text(size = 26, color = "black"),
  ) +
  geom_label_repel( data=subset(don, is_annotate=="yes"), aes(label=Gene), size=7)


print(manhplot)

```


## BF partition


```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

dat <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/BFpartition_misupdated.RDS")

dat_select <- dat[dat$PP > 0.5 & !is.na(dat$PP),]
dat_select <- dat_select[,c("Gene","BF","g1_BF","g2_BF","g3_BF","g4_BF","g5_BF","g6_BF")]
dat_select <- dat_select[order(dat_select$BF, decreasing = T),]

dat_select$logbf <- log2(dat_select$BF)
dat_select$g1_BF <- log2(as.numeric(dat_select$g1_BF))
dat_select$g2_BF <- log2(as.numeric(dat_select$g2_BF))
dat_select$g3_BF <- log2(as.numeric(dat_select$g3_BF))
dat_select$g4_BF <- log2(as.numeric(dat_select$g4_BF))
dat_select$g5_BF <- log2(as.numeric(dat_select$g5_BF))
dat_select$g6_BF <- log2(as.numeric(dat_select$g6_BF))

colnames(dat_select)[3:8] <- cat


dat_select[is.na(dat_select)] <- 0
#dat_select[dat_select < 0] <- 0
dat_select[,2:ncol(dat_select)] <- abs(dat_select[,2:ncol(dat_select)])

# #abs
# numeric_columns <- sapply(dat_select, is.numeric)
# for (col in names(dat_select)[numeric_columns]) {
#   dat_select[[col]] <- ifelse(dat_select[[col]] < 0, abs(dat_select[[col]]), dat_select[[col]])
# }

dat_select$sum_bf <- rowSums(dat_select[,3:8])
dat_select$Gene <- factor(dat_select$Gene, levels = dat_select$Gene)

df_long <- dat_select %>%
  dplyr::select(Gene, highPLI, medPLI, lowPLI, `likely pathogenic`, ambiguous, `likely benign`) %>%
  tidyr::pivot_longer(cols = -Gene, names_to = "variant_type", values_to = "value")


colnames(df_long)[2] <- "Variant Group"
df_long$`Variant Group` <- factor(df_long$`Variant Group`, levels = cat)

# Now plot using ggplot
p <- ggplot(df_long, aes(x = Gene, y = value, fill = `Variant Group`)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(#title = "BF and variant type per gene", 
       y = expression(log[2]*"BF"),
       x = paste(nrow(dat_select),"genes with PP > 0.5")) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 12), 
    axis.text.y = element_text(size = 12), 
    axis.title.x = element_text(size = 14),
    axis.title.y = element_text(size = 14),
  ) +
  #scale_fill_manual(values = c("#B22222", "#FF8C00", "#FFD700","#4682B4","#5F9EA0","#87CEFA"))
  scale_fill_manual(values = palette()[2:7])

print(p)
```

# Validation

## Gene functions

**The bolded ones are the new genes that not identified by initial submission**

| ARAP2       | 243.4156     | 0.927596     | 0.072404     | A protein in Rho-GTPase signaling, implicated in ASD.                                                                                    |
|-------------|--------------|--------------|--------------|------------------------------------------------------------------------------------------------------------------------------------------|
| **PRTG**    | **149.1478** | **0.887004** | **0.0927**   | **Marks early neural progenitors and helps prevent premature neuronal differentiation, reported as novel ASD genes**                     |
| PTPRF       | 148.9174     | 0.886849     | 0.099517     | Protein tyrosine phosphatase family. Synapse formation and differentiation.                                                              |
| **OR52A5**  | **121.3102** | **0.864586** | **0.108491** | **Encodes an olfactory receptor that detects odorants to trigger smell perception.**                                                     |
| PCYOX1L     | 98.35327     | 0.838096     | 0.119174     | Associated with Freemartin-ism, sexual development                                                                                       |
| **LINGO2**  | **78.52239** | **0.805173** | **0.131783** | **Its soluble form promotes excitatory synapse formation and is associated with ASD**                                                    |
| **OR11H6**  | **64.37471** | **0.772113** | **0.145512** | **Encodes an olfactory receptor involved in detecting odorants to initiate smell perception.**                                           |
| MYH10       | 58.56017     | 0.755029     | 0.157944     | Myosin heavy chain. Micro-cephaly, developmental delay, known ASD gene.                                                                  |
| KATNAL1     | 43.80999     | 0.6975       | 0.174006     | Regulates neuronal development, migration, and behavior, with loss-of-function linked to brain abnormalities and ciliary defects in mice |
| LMNB1       | 41.1177      | 0.683953     | 0.18821      | Encodes Lamin B1; essential for neuronal migration; mutations cause CNS disorder                                                         |
| **USP31**   | **38.96737** | **0.672229** | **0.200897** | **Deubiquitinase; regulates NF‑κB signaling and inflammation pathways**                                                                  |
| MFSD10      | 36.81086     | 0.659565     | 0.212526     | Major Facilitator Superfamily transporter; exports organic anions (e.g., NSAIDs)                                                         |
| **OXER1**   | **35.16301** | **0.649207** | **0.223162** | **GPCR for oxoeicosanoids; involved in intracellular Ca²⁺ signaling**                                                                    |
| AHNAK       | 28.92244     | 0.603526     | 0.235541     | Scaffold; regulates neuronal Ca²⁺ channels, BBB, differentiation. Emerging ASD candidate; methylation signal                             |
| **SLC5A6**  | **28.49482** | **0.599956** | **0.246508** | **Sodium-dependent transporter for biotin, pantothenate, lipoate**                                                                       |
| **DYNC1H1** | **24.90788** | **0.567276** | **0.258146** | **Dynein motor protein; retrograde axonal transport, neuronal migration**                                                                |
| **PDS5B**   | **23.19339** | **0.549693** | **0.26945**  | **Cohesin regulator; sister chromatid cohesion, chromatin looping**                                                                      |
| **PTPRE**   | **23.11443** | **0.548848** | **0.279544** | **Receptor-type tyrosine phosphatase; signaling via AKT, ERK, integrins**                                                                |
| PAN2        | 21.83686     | 0.534734     | 0.289319     | mRNA deadenylation                                                                                                                       |
| **OSBPL3**  | **20.67887** | **0.521156** | **0.298796** | **Oxysterol-binding protein family; lipid transport and membrane contact sites**                                                         |
| **SV2B**    | **19.7254**  | **0.509366** | **0.307931** | **Synaptic vesicle glycoprotein; neurotransmitter release modulatio**                                                                    |


## Enrichment results

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

num_topgenes <- sum(gene_pip$PP > 0.5)

load("/project/xinhe/xsun/rare_variants/5.revision/results/enrichment_misupdated.rdata")

df_sum <- df_sum[-which(rownames(df_sum) == "constraint gene"),]
df_sum <- df_sum[-which(rownames(df_sum) == "AutismKB"),]

rownames(df_sum)[which(rownames(df_sum) == "ID gene")] <- "ID"
rownames(df_sum)[which(rownames(df_sum) == "SCZ gene")] <- "SCZ"
rownames(df_sum)[which(rownames(df_sum) == "FMRP")] <- "FMRP target"
rownames(df_sum)[which(rownames(df_sum) == "High")] <- "TADA"
#rownames(df_sum)[which(rownames(df_sum) == "High")] <- "TADA High"
#rownames(df_sum)[which(rownames(df_sum) == "Mod")] <- "TADA Med"
rownames(df_sum)[which(rownames(df_sum) == "SFARI_gene")] <- "SFARI"
rownames(df_sum)[which(rownames(df_sum) == "Haploinsuff")] <- "HI"

#order <- c("SFARI","TADA High","TADA Med","ID","SCZ","FMRP target","PSD","RVIS","HI")
order <- c("SFARI","TADA","ID","SCZ","FMRP target","PSD","RVIS","HI")


df_plot <- df_sum[,c("overlap_rate_mirage","overlap_rate_burden_combined","overlap_rate_burden_lof","overlap_rate_acat","overlap_rate_all")]
colnames(df_plot) <- c(
  sprintf("Top %d genes by MIRAGE", num_topgenes),
  sprintf("Top %d genes by burden test (likely pathogenic&LoF)", num_topgenes),
  sprintf("Top %d genes by burden test (LoF)", num_topgenes),
  sprintf("Top %d genes by ACAT", num_topgenes),
  "All genes"
)
df_plot$category <- rownames(df_plot)

df_fisher <- df_sum[,c(1,15:19)]

df_plot <- df_plot %>%
  mutate(category = as.factor(category)) %>%
  mutate(across(-category, as.numeric))

# Transform the data from wide to long format
long_df <- pivot_longer(df_plot, cols = -category, names_to = "Groups", values_to = "value")
long_df$Groups <- factor(long_df$Groups, levels = colnames(df_plot))

p <- ggplot(long_df, aes(x = factor(category, levels = order), y = value, fill = Groups)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7), width = 0.7) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = "Gene sets", y = "Overlap with Gene sets", fill = "Groups")

print(p)
```



## Enrichment in other gene sets

https://pubmed.ncbi.nlm.nih.gov/31981491/ This paper published several data sets we can use. The gene sets are gtex tissue specific genes, prenatal vs. postnatal gene sets.

Some gene sets were downloaded here https://github.com/BreenMS/Rapid-gene-set-enrichment


```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

gene_pp_bf_posadded_misupdated <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/gene_pp_bf_posadded_misupdated.RDS")

```

### GTEx brain tissue specific genes

We did Fisher exact test to check if the genes with PP>0.5 are enriched in the tissue-specific gene sets. 

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

merged_df <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/validation_gtex_brain_misupdated.RDS")

DT::datatable(merged_df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',' Brain tissue upregulated genes and overlap with MIRAGE genes'),options = list(pageLength = 10) )

merged_df <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/validation_gtex_nonbrain_misupdated.RDS")


DT::datatable(merged_df,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;',' non-Brain tissue upregulated genes and overlap with MIRAGE genes'),options = list(pageLength = 10) )

```

### prenatal, postnatal, unbiased gene sets

We did Fisher exact test to check if the genes with PP>0.5 are enriched in the prenatal/postnatal/unbiased gene sets. 

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

braineffect <-  readxl::read_excel("/project/xinhe/xsun/rare_variants/5.revision/data/brain_de/BrainSpan_FetalEffect.xlsx")

prenatal <- braineffect[braineffect$FetalEffect == "Prenatal",]
unbiased <- braineffect[braineffect$FetalEffect == "Unbiased",]
postnatal <- braineffect[braineffect$FetalEffect == "Postnatal",]

print(paste0("Prenatal gene number =", nrow(prenatal)))
print(paste0("Postnatal gene number =", nrow(postnatal)))
print(paste0("Unbiased gene number =", nrow(unbiased)))

print("Genes with PP > 0.5")
highpip <- gene_pp_bf_posadded_misupdated[gene_pp_bf_posadded_misupdated$PP > 0.5,]

# Define sets
highpip_genes <- highpip$Gene
prenatal_genes <- prenatal$GeneSymbol
postnatal_genes <- postnatal$GeneSymbol
unbiased_genes <- unbiased$GeneSymbol

background_genes <- gene_pp_bf_posadded_misupdated$Gene


# Define `%notin%` for convenience
`%notin%` <- Negate(`%in%`)

print("Prenatal")
print(highpip$Gene[highpip$Gene %in% prenatal$GeneSymbol])
get_fisher_enrichment(prenatal_genes, "Prenatal")

print("Postnatal")
print(highpip$Gene[highpip$Gene %in% postnatal$GeneSymbol])
get_fisher_enrichment(postnatal_genes, "Postnatal")

print("Unbiased")
print(highpip$Gene[highpip$Gene %in% unbiased$GeneSymbol])
get_fisher_enrichment(unbiased_genes, "Unbiased")


```




### Single cell expression data, 

To assess whether the ASD genes are enriched in specific cell types (clusters), we used a background set of 7,867 genes expressed in the experiment. For each cell type, they built a 2×2 table comparing the number of ASD genes expressed versus not expressed, and the number of non-ASD genes expressed versus not expressed. Enrichment was evaluated using odds ratios and Fisher’s exact test.

Within each cell type cluster, a gene was considered expressed if one or more of its transcripts were detected in 25% or more cells

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

fisher_res <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/validation_scrna_fet_misupdated.RDS")

DT::datatable(fisher_res,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Fisher results for the ASD genes'),options = list(pageLength = 10) )

cluster_member <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/validation_scrna_expr_binary_misupdated.RDS")

DT::datatable(cluster_member,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Cluster membership for the ASD genes'),options = list(pageLength = 10) )

```




## Extended gene sets

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}

genesets <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/data/genesets/genesets_summary.RDS")

mirage <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/gene_pp_bf_posadded_misupdated.RDS")

num_topgene <- sum(mirage$PP > 0.5)
mirage_genes <- mirage$Gene[mirage$PP > 0.5]
all_genes <- mirage$Gene

summary <- c()
for (i in 1:length(genesets)){
  
  genes_inset <- genesets[[i]]
  
  num_mirage_gene_in_geneset <- sum(mirage_genes %in% genes_inset)
  
  # Create categories
  mirage_in_set     <- sum(mirage_genes %in% genes_inset)
  mirage_not_in_set <- length(mirage_genes) -  mirage_in_set 
  non_mirage_in_set <- sum(setdiff(all_genes, mirage_genes) %in% genes_inset)
  non_mirage_not_in_set <- length(setdiff(all_genes, mirage_genes)) -  non_mirage_in_set
  
  gene_detail <- mirage_genes[mirage_genes %in% genes_inset]
  
  # Build 2x2 matrix
  contingency_table <- matrix(c(
    mirage_in_set,
    mirage_not_in_set,
    non_mirage_in_set,
    non_mirage_not_in_set
  ), nrow = 2, byrow = TRUE)
  
  colnames(contingency_table) <- c("In_Set", "Not_In_Set")
  rownames(contingency_table) <- c("MIRAGE", "Non_MIRAGE")
  
  # Perform Fisher's Exact Test
  fisher_res <- fisher.test(contingency_table)

  tmp <- c(names(genesets[i]),length(genes_inset), mirage_in_set, paste0(round(mirage_in_set/length(mirage_genes) *100,2), "%"), 
           round(fisher_res$p.value,4), round(fisher_res$estimate,4),paste0(gene_detail,collapse = ","))
  summary <- rbind(summary, tmp)
  
}

summary <- as.data.frame(summary)
colnames(summary) <- c("gene sets", "#of genes in gene set", "#of mirage gene in gene set", "pct of mirage gene in gene set", "FET p", "FET OR", "MIRAGE genes in gene set")

rownames(summary) <- NULL
rownames(summary)[rownames(summary) == "High"] <- "TADA"

DT::datatable(summary,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','FET results for extended gene set'),options = list(pageLength = 10) )



```



# Other methods

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}

source("/project/xinhe/xsun/r_functions/qqplot_multi.R")

burden_pvalues <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/burden_pvalues_misupdated.RDS")
burden_pvalues <- as.data.frame(burden_pvalues)

ACAT <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/acat_misupdated.RDS")


pvalues <- list('Burden test (LoF)' = burden_pvalues$LoF_pvalue,
                'Burden test (likely pathogenic &LoF)' = burden_pvalues$fisher_combine_pvalue, 
                'ACAT' = ACAT$ACAT_pvalue)

p <- qqplot_multi(pvalues_list = pvalues,legend_names = c('Burden test (LoF)','Burden test (likely pathogenic &LoF)','ACAT'))

print(p)
```



# Comparing with results in initial submission

```{r fig.height=3, fig.width=8 , warning=F, message=FALSE}

mirage_res_old <- read.csv("/project2/xinhe/xsun/rare_variants/4.additional_analysis/data/data_manhattan_plot_AllInheritedVariant_deltafixed_20231102.csv")

compare <- merge(gene_pip, mirage_res_old, by = "Gene")
colnames(compare) <- c("Gene","BF_new", "post_prob_new","chr","pos" ,"PP_new","FDR_new","X","BF_old","post_prob_old","chr.y","pos.y")

compare <- compare[,c("Gene","chr","pos","BF_new", "post_prob_new","FDR_new","BF_old","post_prob_old")] 
compare$pp_diff <- abs(compare$post_prob_new - compare$post_prob_old)

# Set a threshold for labeling (adjustable)
label_threshold <- 0.3
compare$label <- ifelse(compare$pp_diff > label_threshold, compare$Gene, NA)


p1 <- ggplot(compare, aes(x = BF_old, y = BF_new)) +
  geom_point(alpha = 0.7, color = "darkorange") +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray40") +
  geom_text_repel(aes(label = label), size = 2.5, max.overlaps = 100) +
  labs(
    title = "PP diff > 0.3 are labeled",
    x = "BF_old",
    y = "BF_new"
  ) +
  theme_minimal()

# Plot 2: post_prob_new vs post_prob_old
p2 <- ggplot(compare, aes(x = post_prob_old, y = post_prob_new)) +
    geom_point(alpha = 0.7, color = "darkorange") +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray40") +
    geom_text_repel(aes(label = label), size = 2.5, max.overlaps = 100) +
    labs(
        title = "PP diff > 0.3 are labeled",
        x = "PP_old",
        y = "PP_new"
    ) +
    theme_minimal()

grid.arrange(p1, p2,  ncol = 2)

genes_diff <- compare[!is.na(compare$label),]

cat_compare <- c("highPLI", "medPLI", "lowPLI", "highMPC/likely pathogenic", "medMPC/ambiguous", "lowMPC/likely benign")
mirage_input_old <- data.table::fread("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_20231013.txt")

for (i in 1:nrow(genes_diff)){
  
  gene <- genes_diff$Gene[i]
  mirage_input_new_gene <- mirage_input_mis_updated[mirage_input_mis_updated$Gene == gene,]
  table_new <- table(mirage_input_new_gene$group.index)[c("1","2","3","4","5","6")]
  names(table_new) <- cat_compare
 
  print(paste0("Input for gene ",gene))

  mirage_input_old_gene <- mirage_input_old[mirage_input_old$Gene == gene,]
  table_old <- table(mirage_input_old_gene$group.index)[c("1","2","3","4","5","6")]
  names(table_old) <- cat_compare

  table_2 <- rbind(table_new,table_old)
  rownames(table_2) <- c("input_new","input_old")
  print(table_2)
  
}


```
