---
title: "MIRAGE results -- missense annotation updated"
author: "XSun"
date: "2025-06-23"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

```{r, warning=F, message=FALSE}
library(ggplot2)
library(dplyr)
library(ggrepel)
library(data.table)
library(tidyr)
library(gridExtra)

mirage_res <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/mirage_misupdated.RDS")

#cat <- c("high LOEUF oe", "med LOEUF oe", "low LOEUF oe", "likely pathogenic", "ambiguous", "likely benign")
cat <- c("highPLI", "medPLI", "lowPLI", "likely pathogenic", "ambiguous", "likely benign")
cat_old <- c("highPLI", "medPLI", "lowPLI", "highMPC", "medMPC", "lowMPC")
```

# Overview of input

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}

input <- fread("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_20231013.txt")
print("Old mirage input:")

sprintf("Number of rare-variants = %s",nrow(input))
old_mirage <- table(input$group.index)[c("1","2","3","4","5","6")]
names(old_mirage) <- c("highPLI", "medPLI", "lowPLI", "highMPC", "medMPC", "lowMPC")
print(old_mirage)

print("Missense updated mirage input:")
mirage_input_mis_updated <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_misupdated.RDS")
new_mirage <- table(mirage_input_mis_updated$group.index)[c("1","2","3","4","5","6")]
names(new_mirage) <- cat
sprintf("Number of rare-variants = %s",nrow(mirage_input_mis_updated))
print(new_mirage)
```


# Parameter estimation

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}
est <- mirage_res$eta.est[c("1","2","3","4","5","6")]
names(mirage_res$eta.pvalue) <- names(mirage_res$eta.est)
pval <- mirage_res$eta.pvalue[c("1","2","3","4","5","6")]


order <- c(1:length(est))

df <- data.frame(est = est, pval = pval, cat = cat, order=order)
df <- df[sort(df$order,decreasing = T),]
df$cat <- factor(df$cat, levels = cat)

df$fill <- ifelse(df$order <= 3, "orange", "steelblue")
df$legend <- ifelse(df$order <= 3, "LoF", "Missense")

# Convert the legend column to a factor and set the levels
df$legend <- factor(df$legend, levels = c("LoF", "Missense"))

p <- ggplot(df, aes(x = cat, y = est, fill = legend)) +
  geom_bar(stat = "identity", width = 0.6) + # Use the fill aesthetic
  geom_text(aes(label = ifelse(pval < 0.05, "*", "")), vjust = 0, size = 6) +
  annotate("text", x = 6, y = 0.9, label = "*: p-value < 0.05", hjust = "right", size = 6) +
  theme_minimal() +
  theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
    axis.text.y = element_text(size = 12),
    legend.title = element_blank() # Hide the legend title
  ) +
  scale_y_continuous(breaks = seq(0, 1, 0.2)) +
  scale_fill_manual(values = c("LoF" = "orange", "Missense" = "steelblue")) # Define manual colors for the legend

print(p)

```


# Gene-level results

## Posterior probability, FDR 

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

gene_pip <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/gene_pp_bf_posadded_misupdated.RDS")

DT::datatable(gene_pip,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Gene-level results'),options = list(pageLength = 10) )

```


## Manhattan plot

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

pp_threshold <- 0.7

don <- gene_pip %>% 
  
  # Compute chromosome size
  group_by(chr) %>% 
  summarise(chr_len=max(pos)) %>% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(gene_pip, ., by=c("chr"="chr")) %>%
  
  # Add a cumulative position of each SNP
  arrange(chr, pos) %>%
  mutate( poscum=pos+tot) %>%
  mutate( is_annotate=ifelse(Gene %in% gene_pip$Gene[gene_pip$post_prob > pp_threshold], "yes", "no"))

# Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
axisdf = don %>% group_by(chr) %>% summarize(center=( max(poscum) + min(poscum) ) / 2 )

manhplot <- ggplot(don, aes(x=poscum, y=post_prob)) +
  
  # Show all points
  geom_point( aes(color=as.factor(chr)), alpha=2, size=3) +
  scale_color_manual(values = rep(c("grey", "steelblue"), 22 )) +
  
  # custom X axis:
  scale_x_continuous( label = axisdf$chr, breaks= axisdf$center ) +
  #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  
  ylab("Posterior Probability") +
  #ylim(0,max(log10(don$BF)) + 2) +
  ylim(0,1) +
  xlab("Chromosome") +
  
  geom_hline(yintercept = pp_threshold, col = "red") +
  annotate("text", x = Inf, y = pp_threshold, label = sprintf("PP = %s", pp_threshold), hjust = 1.1, vjust = -1, color = "red", size = 7) +
  
  # Custom the theme:
  theme_bw() +
  #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
  theme( 
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_text(size = 28),
    axis.text.x = element_text(size = 26, color = "black"),
    axis.title.y = element_text(size = 28),
    axis.text.y = element_text(size = 26, color = "black"),
  ) +
  geom_label_repel( data=subset(don, is_annotate=="yes"), aes(label=Gene), size=7)


print(manhplot)

```


## BF partition


```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

dat <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/BFpartition_misupdated.RDS")

dat_select <- dat[dat$PP > 0.5 & !is.na(dat$PP),]
dat_select <- dat_select[,c("Gene","BF","g1_BF","g2_BF","g3_BF","g4_BF","g5_BF","g6_BF")]
dat_select <- dat_select[order(dat_select$BF, decreasing = T),]

dat_select$logbf <- log2(dat_select$BF)
dat_select$g1_BF <- log2(as.numeric(dat_select$g1_BF))
dat_select$g2_BF <- log2(as.numeric(dat_select$g2_BF))
dat_select$g3_BF <- log2(as.numeric(dat_select$g3_BF))
dat_select$g4_BF <- log2(as.numeric(dat_select$g4_BF))
dat_select$g5_BF <- log2(as.numeric(dat_select$g5_BF))
dat_select$g6_BF <- log2(as.numeric(dat_select$g6_BF))

colnames(dat_select)[3:8] <- cat


dat_select[is.na(dat_select)] <- 0
#dat_select[dat_select < 0] <- 0
dat_select[,2:ncol(dat_select)] <- abs(dat_select[,2:ncol(dat_select)])

# #abs
# numeric_columns <- sapply(dat_select, is.numeric)
# for (col in names(dat_select)[numeric_columns]) {
#   dat_select[[col]] <- ifelse(dat_select[[col]] < 0, abs(dat_select[[col]]), dat_select[[col]])
# }

dat_select$sum_bf <- rowSums(dat_select[,3:8])
dat_select$Gene <- factor(dat_select$Gene, levels = dat_select$Gene)

df_long <- dat_select %>%
  dplyr::select(Gene, highPLI, medPLI, lowPLI, `likely pathogenic`, ambiguous, `likely benign`) %>%
  tidyr::pivot_longer(cols = -Gene, names_to = "variant_type", values_to = "value")


colnames(df_long)[2] <- "Variant Group"
df_long$`Variant Group` <- factor(df_long$`Variant Group`, levels = cat)

# Now plot using ggplot
p <- ggplot(df_long, aes(x = Gene, y = value, fill = `Variant Group`)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(#title = "BF and variant type per gene", 
       y = expression(log[2]*"BF"),
       x = paste(nrow(dat_select),"genes with PP > 0.5")) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 12), 
    axis.text.y = element_text(size = 12), 
    axis.title.x = element_text(size = 14),
    axis.title.y = element_text(size = 14),
  ) +
  #scale_fill_manual(values = c("#B22222", "#FF8C00", "#FFD700","#4682B4","#5F9EA0","#87CEFA"))
  scale_fill_manual(values = palette()[2:7])

print(p)
```

# Validation 

## Enrichment results

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

num_topgenes <- sum(gene_pip$PP > 0.5)

load("/project/xinhe/xsun/rare_variants/5.revision/results/enrichment_misupdated.rdata")

df_sum <- df_sum[-which(rownames(df_sum) == "constraint gene"),]
df_sum <- df_sum[-which(rownames(df_sum) == "AutismKB"),]

rownames(df_sum)[which(rownames(df_sum) == "ID gene")] <- "ID"
rownames(df_sum)[which(rownames(df_sum) == "SCZ gene")] <- "SCZ"
rownames(df_sum)[which(rownames(df_sum) == "FMRP")] <- "FMRP target"
rownames(df_sum)[which(rownames(df_sum) == "High")] <- "TADA"
#rownames(df_sum)[which(rownames(df_sum) == "High")] <- "TADA High"
#rownames(df_sum)[which(rownames(df_sum) == "Mod")] <- "TADA Med"
rownames(df_sum)[which(rownames(df_sum) == "SFARI_gene")] <- "SFARI"
rownames(df_sum)[which(rownames(df_sum) == "Haploinsuff")] <- "HI"

#order <- c("SFARI","TADA High","TADA Med","ID","SCZ","FMRP target","PSD","RVIS","HI")
order <- c("SFARI","TADA","ID","SCZ","FMRP target","PSD","RVIS","HI")


df_plot <- df_sum[,c("overlap_rate_mirage","overlap_rate_burden_combined","overlap_rate_burden_lof","overlap_rate_acat","overlap_rate_all")]
colnames(df_plot) <- c(
  sprintf("Top %d genes by MIRAGE", num_topgenes),
  sprintf("Top %d genes by burden test (likely pathogenic&LoF)", num_topgenes),
  sprintf("Top %d genes by burden test (LoF)", num_topgenes),
  sprintf("Top %d genes by ACAT", num_topgenes),
  "All genes"
)
df_plot$category <- rownames(df_plot)

df_fisher <- df_sum[,c(1,15:19)]

df_plot <- df_plot %>%
  mutate(category = as.factor(category)) %>%
  mutate(across(-category, as.numeric))

# Transform the data from wide to long format
long_df <- pivot_longer(df_plot, cols = -category, names_to = "Groups", values_to = "value")
long_df$Groups <- factor(long_df$Groups, levels = colnames(df_plot))

p <- ggplot(long_df, aes(x = factor(category, levels = order), y = value, fill = Groups)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7), width = 0.7) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = "Gene sets", y = "Overlap with Gene sets", fill = "Groups")

print(p)
```

# Other methods

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}

source("/project/xinhe/xsun/r_functions/qqplot_multi.R")

burden_pvalues <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/burden_pvalues_misupdated.RDS")
burden_pvalues <- as.data.frame(burden_pvalues)

ACAT <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/acat_misupdated.RDS")


pvalues <- list('Burden test (LoF)' = burden_pvalues$LoF_pvalue,
                'Burden test (likely pathogenic &LoF)' = burden_pvalues$fisher_combine_pvalue, 
                'ACAT' = ACAT$ACAT_pvalue)

p <- qqplot_multi(pvalues_list = pvalues,legend_names = c('Burden test (LoF)','Burden test (likely pathogenic &LoF)','ACAT'))

print(p)
```



# Comparing with results in initial submission

```{r fig.height=3, fig.width=8 , warning=F, message=FALSE}

mirage_res_old <- read.csv("/project2/xinhe/xsun/rare_variants/4.additional_analysis/data/data_manhattan_plot_AllInheritedVariant_deltafixed_20231102.csv")

compare <- merge(gene_pip, mirage_res_old, by = "Gene")
colnames(compare) <- c("Gene","BF_new", "post_prob_new","chr","pos" ,"PP_new","FDR_new","X","BF_old","post_prob_old","chr.y","pos.y")

compare <- compare[,c("Gene","chr","pos","BF_new", "post_prob_new","FDR_new","BF_old","post_prob_old")] 
compare$pp_diff <- abs(compare$post_prob_new - compare$post_prob_old)

# Set a threshold for labeling (adjustable)
label_threshold <- 0.3
compare$label <- ifelse(compare$pp_diff > label_threshold, compare$Gene, NA)


p1 <- ggplot(compare, aes(x = BF_old, y = BF_new)) +
  geom_point(alpha = 0.7, color = "darkorange") +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray40") +
  geom_text_repel(aes(label = label), size = 2.5, max.overlaps = 100) +
  labs(
    title = "PP diff > 0.3 are labeled",
    x = "BF_old",
    y = "BF_new"
  ) +
  theme_minimal()

# Plot 2: post_prob_new vs post_prob_old
p2 <- ggplot(compare, aes(x = post_prob_old, y = post_prob_new)) +
    geom_point(alpha = 0.7, color = "darkorange") +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray40") +
    geom_text_repel(aes(label = label), size = 2.5, max.overlaps = 100) +
    labs(
        title = "PP diff > 0.3 are labeled",
        x = "PP_old",
        y = "PP_new"
    ) +
    theme_minimal()

grid.arrange(p1, p2,  ncol = 2)

genes_diff <- compare[!is.na(compare$label),]

cat_compare <- c("highPLI", "medPLI", "lowPLI", "highMPC/likely pathogenic", "medMPC/ambiguous", "lowMPC/likely benign")
mirage_input_old <- data.table::fread("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_20231013.txt")

for (i in 1:nrow(genes_diff)){
  
  gene <- genes_diff$Gene[i]
  mirage_input_new_gene <- mirage_input_mis_updated[mirage_input_mis_updated$Gene == gene,]
  table_new <- table(mirage_input_new_gene$group.index)[c("1","2","3","4","5","6")]
  names(table_new) <- cat_compare
 
  print(paste0("Input for gene ",gene))

  mirage_input_old_gene <- mirage_input_old[mirage_input_old$Gene == gene,]
  table_old <- table(mirage_input_old_gene$group.index)[c("1","2","3","4","5","6")]
  names(table_old) <- cat_compare

  table_2 <- rbind(table_new,table_old)
  rownames(table_2) <- c("input_new","input_old")
  print(table_2)
  
}


```
