---
title: "MIRAGE results -- lof annotation updated  -- lof use decile"
author: "XSun"
date: "2025-08-04"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---

```{r, warning=F, message=FALSE}
library(ggplot2)
library(dplyr)
library(ggrepel)
library(data.table)
library(tidyr)
library(gridExtra)

mirage_res <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/mirage_lofdeconly.RDS")

cat <- c("low LOEUF oe", "med LOEUF oe", "high LOEUF oe", "highMPC", "medMPC", "lowMPC")
# cat <- c("highPLI", "medPLI", "lowPLI", "likely pathogenic", "ambiguous", "likely benign")
cat_old <- c("highPLI", "medPLI", "lowPLI", "highMPC", "medMPC", "lowMPC")
```

# Overview of input

LoF Annotation Grouping (by Decile):

Group 1: LoF decile = 1
Group 2: LoF decile = 2 or 3
Group 3: LoF decile = 4 to 10

```
quantile(mirage_input_updated$lof, probs = seq(0, 1, 0.1), na.rm = TRUE)
0%      10%      20%      30%      40%      50%      60%      70%      80%      90%     100% 
0.000000 0.437728 0.563360 0.648770 0.713100 0.770100 0.826110 0.887400 0.961480 1.083500 4.166300 
```


```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}

input <- fread("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_20231013.txt")
print("Old mirage input:")

sprintf("Number of rare-variants = %s",nrow(input))
old_mirage <- table(input$group.index)[c("1","2","3","4","5","6")]
names(old_mirage) <- c("highPLI", "medPLI", "lowPLI", "highMPC", "medMPC", "lowMPC")
print(old_mirage)

print("LoF & Missense updated mirage input:")
mirage_input_mis_updated <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_lofdeconly.RDS")
new_mirage <- table(mirage_input_mis_updated$group.index)[c("1","2","3","4","5","6")]
names(new_mirage) <- cat
sprintf("Number of rare-variants = %s",nrow(mirage_input_mis_updated))
print(new_mirage)
```


# Parameter estimation

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}
est <- mirage_res$eta.est[c("1","2","3","4","5","6")]
names(mirage_res$eta.pvalue) <- names(mirage_res$eta.est)
pval <- mirage_res$eta.pvalue[c("1","2","3","4","5","6")]


order <- c(1:length(est))

df <- data.frame(est = est, pval = pval, cat = cat, order=order)
df <- df[sort(df$order,decreasing = T),]
df$cat <- factor(df$cat, levels = cat)

df$fill <- ifelse(df$order <= 3, "orange", "steelblue")
df$legend <- ifelse(df$order <= 3, "LoF", "Missense")

# Convert the legend column to a factor and set the levels
df$legend <- factor(df$legend, levels = c("LoF", "Missense"))

p <- ggplot(df, aes(x = cat, y = est, fill = legend)) +
  geom_bar(stat = "identity", width = 0.6) + # Use the fill aesthetic
  geom_text(aes(label = ifelse(pval < 0.05, "*", "")), vjust = 0, size = 6) +
  annotate("text", x = 6, y = 0.9, label = "*: p-value < 0.05", hjust = "right", size = 6) +
  theme_minimal() +
  theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 12),
    axis.text.y = element_text(size = 12),
    legend.title = element_blank() # Hide the legend title
  ) +
  scale_y_continuous(breaks = seq(0, 1, 0.2)) +
  scale_fill_manual(values = c("LoF" = "orange", "Missense" = "steelblue")) # Define manual colors for the legend

print(p)

```


# Gene-level results

## Posterior probability, FDR 

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

gene_pip <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/gene_pp_bf_posadded_lofdeconly.RDS")

DT::datatable(gene_pip,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','Gene-level results'),options = list(pageLength = 10) )

```


## Manhattan plot

```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

pp_threshold <- 0.7

don <- gene_pip %>% 
  
  # Compute chromosome size
  group_by(chr) %>% 
  summarise(chr_len=max(pos)) %>% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(as.numeric(chr_len))-as.numeric(chr_len)) %>%
  dplyr::select(-chr_len) %>%
  
  # Add this info to the initial dataset
  left_join(gene_pip, ., by=c("chr"="chr")) %>%
  
  # Add a cumulative position of each SNP
  arrange(chr, pos) %>%
  mutate( poscum=pos+tot) %>%
  mutate( is_annotate=ifelse(Gene %in% gene_pip$Gene[gene_pip$post_prob > pp_threshold], "yes", "no"))

# Then we need to prepare the X axis. Indeed we do not want to display the cumulative position of SNP in bp, but just show the chromosome name instead.
axisdf = don %>% group_by(chr) %>% summarize(center=( max(poscum) + min(poscum) ) / 2 )

manhplot <- ggplot(don, aes(x=poscum, y=post_prob)) +
  
  # Show all points
  geom_point( aes(color=as.factor(chr)), alpha=2, size=3) +
  scale_color_manual(values = rep(c("grey", "steelblue"), 22 )) +
  
  # custom X axis:
  scale_x_continuous( label = axisdf$chr, breaks= axisdf$center ) +
  #scale_y_continuous(expand = c(0, 0) ) +     # remove space between plot area and x axis
  
  ylab("Posterior Probability") +
  #ylim(0,max(log10(don$BF)) + 2) +
  ylim(0,1) +
  xlab("Chromosome") +
  
  geom_hline(yintercept = pp_threshold, col = "red") +
  annotate("text", x = Inf, y = pp_threshold, label = sprintf("PP = %s", pp_threshold), hjust = 1.1, vjust = -1, color = "red", size = 7) +
  
  # Custom the theme:
  theme_bw() +
  #ggtitle(i) + theme(plot.title = element_text(hjust = 0.5)) +
  theme( 
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.title.x = element_text(size = 28),
    axis.text.x = element_text(size = 26, color = "black"),
    axis.title.y = element_text(size = 28),
    axis.text.y = element_text(size = 26, color = "black"),
  ) +
  geom_label_repel( data=subset(don, is_annotate=="yes"), aes(label=Gene), size=7)


print(manhplot)

```


## BF partition


```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

dat <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/BFpartition_lofdeconly.RDS")

dat_select <- dat[dat$PP > 0.5 & !is.na(dat$PP),]
dat_select <- dat_select[,c("Gene","BF","g1_BF","g2_BF","g3_BF","g4_BF","g5_BF","g6_BF")]
dat_select <- dat_select[order(dat_select$BF, decreasing = T),]

dat_select$logbf <- log2(dat_select$BF)
dat_select$g1_BF <- log2(as.numeric(dat_select$g1_BF))
dat_select$g2_BF <- log2(as.numeric(dat_select$g2_BF))
dat_select$g3_BF <- log2(as.numeric(dat_select$g3_BF))
dat_select$g4_BF <- log2(as.numeric(dat_select$g4_BF))
dat_select$g5_BF <- log2(as.numeric(dat_select$g5_BF))
dat_select$g6_BF <- log2(as.numeric(dat_select$g6_BF))

colnames(dat_select)[3:8] <- cat


dat_select[is.na(dat_select)] <- 0
#dat_select[dat_select < 0] <- 0
dat_select[,2:ncol(dat_select)] <- abs(dat_select[,2:ncol(dat_select)])

# #abs
# numeric_columns <- sapply(dat_select, is.numeric)
# for (col in names(dat_select)[numeric_columns]) {
#   dat_select[[col]] <- ifelse(dat_select[[col]] < 0, abs(dat_select[[col]]), dat_select[[col]])
# }

dat_select$sum_bf <- rowSums(dat_select[,3:8])
dat_select$Gene <- factor(dat_select$Gene, levels = dat_select$Gene)

df_long <- dat_select %>%
  dplyr::select(Gene, `low LOEUF oe`, `med LOEUF oe`, `high LOEUF oe`,  highMPC, medMPC, lowMPC) %>%
  tidyr::pivot_longer(cols = -Gene, names_to = "variant_type", values_to = "value")


colnames(df_long)[2] <- "Variant Group"
df_long$`Variant Group` <- factor(df_long$`Variant Group`, levels = cat)

# Now plot using ggplot
p <- ggplot(df_long, aes(x = Gene, y = value, fill = `Variant Group`)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(#title = "BF and variant type per gene", 
       y = expression(log[2]*"BF"),
       x = paste(nrow(dat_select),"genes with PP > 0.5")) +
  theme_minimal() +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, size = 12), 
    axis.text.y = element_text(size = 12), 
    axis.title.x = element_text(size = 14),
    axis.title.y = element_text(size = 14),
  ) +
  #scale_fill_manual(values = c("#B22222", "#FF8C00", "#FFD700","#4682B4","#5F9EA0","#87CEFA"))
  scale_fill_manual(values = palette()[2:7])

print(p)
```



# Validation 

## Gene functions

**The bolded ones are the new genes that not identified by initial submission**

| Gene        | BF           | PP           | FDR          | Functions                                                                        |
|-------------|--------------|--------------|--------------|----------------------------------------------------------------------------------|
| PLXNA1      | 316.6647     | 0.943396     | 0.056604     | Transmembrane receptor. Essential for axon guidance.                             |
| PCYOX1L     | 105.796      | 0.847752     | 0.104426     | Associated with Freemartinism, sexual development                                |
| MYH10       | 89.17553     | 0.82436      | 0.128164     | Myosin heavy chain. Microcephaly, developmental delay, known ASD gene.           |
| ARAP2       | 61.20125     | 0.763096     | 0.155349     | A protein in Rho-GTPase signaling, implicated in ASD.                            |
| LMNB1       | 60.5662      | 0.761205     | 0.172038     | Encodes Lamin B1; essential for neuronal migration; mutations cause CNS disorder |
| **SLC5A6**  | **45.88393** | **0.707169** | **0.19217**  | **Biotin/pantothenic acid transport across blood brain barrier**                 |
| PTPRF       | 45.8515      | 0.707023     | 0.206571     | Protein tyrosine phosphatase family. Synapse formation and differentiation.      |
| **UBR1**    | **44.55561** | **0.701049** | **0.218119** | **N-end rule E3 ubiquitin ligase; protein QC**                                   |
| BTG3        | 40.35893     | 0.679913     | 0.229449     | Antiproliferative; neurogenesis potential                                        |
| SF3B3       | 38.25422     | 0.668147     | 0.239689     | Pre‑mRNA splicing factor                                                         |
| **TRAK1**   | **33.86689** | **0.640607** | **0.250571** | **Mitochondrial transport in neurons**                                           |
| SYNGAP1     | 32.81031     | 0.633278     | 0.260251     | Synaptic Ras‑GAP; synaptic plasticity                                            |
| IL1R2       | 29.27636     | 0.606433     | 0.270506     | Decoy IL‑1 receptor regulating inflammation                                      |
| **SV2B**    | **28.12839** | **0.596846** | **0.279981** | **Synaptic vesicle release modulation**                                          |
| NAV3        | 27.84933     | 0.594445     | 0.288352     | Axon guidance / neuronal migration                                               |
| MYO5C       | 26.66308     | 0.583909     | 0.296336     | Motor protein for vesicle/organelle transport                                    |
| PAN2        | 25.81673     | 0.576051     | 0.303843     | mRNA deadenylation                                                               |
| ATP2B2      | 25.72282     | 0.575161     | 0.310565     | Ca²⁺ extrusion pump (PMCA isoform)                                               |
| **NCKAP5L** | **23.29762** | **0.550802** | **0.317861** | **Actin regulation; potential neural role**                                      |


## Enrichment results

### gene sets in initial submission 


```{r fig.height=5, fig.width=9, warning=F, message=FALSE}

num_topgenes <- sum(gene_pip$PP > 0.5)

load("/project/xinhe/xsun/rare_variants/5.revision/results/enrichment_lofdeconly.rdata")

df_sum <- df_sum[-which(rownames(df_sum) == "constraint gene"),]
df_sum <- df_sum[-which(rownames(df_sum) == "AutismKB"),]

rownames(df_sum)[which(rownames(df_sum) == "ID gene")] <- "ID"
rownames(df_sum)[which(rownames(df_sum) == "SCZ gene")] <- "SCZ"
rownames(df_sum)[which(rownames(df_sum) == "FMRP")] <- "FMRP target"
rownames(df_sum)[which(rownames(df_sum) == "High")] <- "TADA"
#rownames(df_sum)[which(rownames(df_sum) == "High")] <- "TADA High"
#rownames(df_sum)[which(rownames(df_sum) == "Mod")] <- "TADA Med"
rownames(df_sum)[which(rownames(df_sum) == "SFARI_gene")] <- "SFARI"
rownames(df_sum)[which(rownames(df_sum) == "Haploinsuff")] <- "HI"

#order <- c("SFARI","TADA High","TADA Med","ID","SCZ","FMRP target","PSD","RVIS","HI")
order <- c("SFARI","TADA","ID","SCZ","FMRP target","PSD","RVIS","HI")


df_plot <- df_sum[,c("overlap_rate_mirage","overlap_rate_burden_combined","overlap_rate_burden_lof","overlap_rate_acat","overlap_rate_all")]
#colnames(df_plot) <- c("Top 20 genes by MIRAGE","Top 20 genes by burden test (MPC>2&LoF)","Top 20 genes by burden test (LoF)","Top 20 genes by ACAT","All genes")
colnames(df_plot) <- c(
  sprintf("Top %d genes by MIRAGE", num_topgenes),
  sprintf("Top %d genes by burden test (likely pathogenic &LoF)", num_topgenes),
  sprintf("Top %d genes by burden test (LoF)", num_topgenes),
  sprintf("Top %d genes by ACAT", num_topgenes),
  "All genes"
)
df_plot$category <- rownames(df_plot)

df_fisher <- df_sum[,c(1,15:19)]

df_plot <- df_plot %>%
  mutate(category = as.factor(category)) %>%
  mutate(across(-category, as.numeric))

# Transform the data from wide to long format
long_df <- pivot_longer(df_plot, cols = -category, names_to = "Groups", values_to = "value")
long_df$Groups <- factor(long_df$Groups, levels = colnames(df_plot))

p <- ggplot(long_df, aes(x = factor(category, levels = order), y = value, fill = Groups)) +
  geom_bar(stat = "identity", position = position_dodge(width = 0.7), width = 0.7) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = "Gene sets", y = "Overlap with Gene sets", fill = "Groups")

print(p)
```






### Extended gene sets

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}

genesets <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/data/genesets/genesets_summary.RDS")

mirage <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/mirage_res/gene_pp_bf_posadded_lofdeconly.RDS")

num_topgene <- sum(mirage$PP > 0.5)
mirage_genes <- mirage$Gene[mirage$PP > 0.5]
all_genes <- mirage$Gene

summary <- c()
for (i in 1:length(genesets)){
  
  genes_inset <- genesets[[i]]
  
  num_mirage_gene_in_geneset <- sum(mirage_genes %in% genes_inset)
  
  # Create categories
  mirage_in_set     <- sum(mirage_genes %in% genes_inset)
  mirage_not_in_set <- length(mirage_genes) -  mirage_in_set 
  non_mirage_in_set <- sum(setdiff(all_genes, mirage_genes) %in% genes_inset)
  non_mirage_not_in_set <- length(setdiff(all_genes, mirage_genes)) -  non_mirage_in_set
  
  gene_detail <- mirage_genes[mirage_genes %in% genes_inset]
  
  # Build 2x2 matrix
  contingency_table <- matrix(c(
    mirage_in_set,
    mirage_not_in_set,
    non_mirage_in_set,
    non_mirage_not_in_set
  ), nrow = 2, byrow = TRUE)
  
  colnames(contingency_table) <- c("In_Set", "Not_In_Set")
  rownames(contingency_table) <- c("MIRAGE", "Non_MIRAGE")
  
  # Perform Fisher's Exact Test
  fisher_res <- fisher.test(contingency_table)

  tmp <- c(names(genesets[i]),length(genes_inset), mirage_in_set, paste0(round(mirage_in_set/length(mirage_genes) *100,2), "%"), 
           round(fisher_res$p.value,4), round(fisher_res$estimate,4),paste0(gene_detail,collapse = ","))
  summary <- rbind(summary, tmp)
  
}

summary <- as.data.frame(summary)
colnames(summary) <- c("gene sets", "#of genes in gene set", "#of mirage gene in gene set", "pct of mirage gene in gene set", "FET p", "FET OR", "MIRAGE genes in gene set")

rownames(summary) <- NULL
rownames(summary)[rownames(summary) == "High"] <- "TADA"

DT::datatable(summary,caption = htmltools::tags$caption( style = 'caption-side: left; text-align: left; color:black;  font-size:150% ;','FET results for extended gene set'),options = list(pageLength = 10) )



```


# Other methods

```{r fig.height=3, fig.width=5 , warning=F, message=FALSE}

source("/project/xinhe/xsun/r_functions/qqplot_multi.R")

burden_pvalues <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/burden_pvalues_lofdeconly.RDS")
burden_pvalues <- as.data.frame(burden_pvalues)

ACAT <- readRDS("/project/xinhe/xsun/rare_variants/5.revision/results/acat_lofdeconly.RDS")


pvalues <- list('Burden test (LoF)' = burden_pvalues$LoF_pvalue,
                'Burden test (likely pathogenic & LoF)' = burden_pvalues$fisher_combine_pvalue, 
                'ACAT' = ACAT$ACAT_pvalue)

p <- qqplot_multi(pvalues_list = pvalues,legend_names = c('Burden test (LoF)','Burden test (likely pathogenic & LoF)','ACAT'))

print(p)
```


# Comparing with results in initial submission

```{r fig.height=3, fig.width=8 , warning=F, message=FALSE}

mirage_res_old <- read.csv("/project2/xinhe/xsun/rare_variants/4.additional_analysis/data/data_manhattan_plot_AllInheritedVariant_deltafixed_20231102.csv")

compare <- merge(gene_pip, mirage_res_old, by = "Gene")
colnames(compare) <- c("Gene","BF_new", "post_prob_new","chr","pos" ,"PP_new","FDR_new","X","BF_old","post_prob_old","chr.y","pos.y")

compare <- compare[,c("Gene","chr","pos","BF_new", "post_prob_new","FDR_new","BF_old","post_prob_old")] 
compare$pp_diff <- abs(compare$post_prob_new - compare$post_prob_old)

# Set a threshold for labeling (adjustable)
label_threshold <- 0.3
compare$label <- ifelse(compare$pp_diff > label_threshold, compare$Gene, NA)


p1 <- ggplot(compare, aes(x = BF_old, y = BF_new)) +
  geom_point(alpha = 0.7, color = "darkorange") +
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray40") +
  geom_text_repel(aes(label = label), size = 2.5, max.overlaps = 100) +
  labs(
    title = "PP diff > 0.3 are labeled",
    x = "BF_old",
    y = "BF_new"
  ) +
  theme_minimal()

# Plot 2: post_prob_new vs post_prob_old
p2 <- ggplot(compare, aes(x = post_prob_old, y = post_prob_new)) +
    geom_point(alpha = 0.7, color = "darkorange") +
    geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray40") +
    geom_text_repel(aes(label = label), size = 2.5, max.overlaps = 100) +
    labs(
        title = "PP diff > 0.3 are labeled",
        x = "PP_old",
        y = "PP_new"
    ) +
    theme_minimal()

grid.arrange(p1, p2,  ncol = 2)

genes_diff <- compare[!is.na(compare$label),]


cat_compare <- c("highPLI", "medPLI", "lowPLI", "highMPC/likely pathogenic", "medMPC/ambiguous", "lowMPC/likely benign")
mirage_input_old <- data.table::fread("/project/xinhe/xsun/rare_variants/5.revision/data/mirage_input_20231013.txt")

for (i in 1:nrow(genes_diff)){
  
  gene <- genes_diff$Gene[i]
  mirage_input_new_gene <- mirage_input_mis_updated[mirage_input_mis_updated$Gene == gene,]
  table_new <- table(mirage_input_new_gene$group.index)[c("1","2","3","4","5","6")]
  names(table_new) <- cat_compare
 
  print(paste0("Input for gene ",gene))

  mirage_input_old_gene <- mirage_input_old[mirage_input_old$Gene == gene,]
  table_old <- table(mirage_input_old_gene$group.index)[c("1","2","3","4","5","6")]
  names(table_old) <- cat_compare

  table_2 <- rbind(table_new,table_old)
  rownames(table_2) <- c("input_new","input_old")
  print(table_2)
  
}


```